### PR TITLE
FYLE-2vgfnd: Tfc revamp - Expense page and service method changes

### PR DESCRIPTION
**Pull request by Dimple16**

### CODE REVIEW

File: src/app/core/services/expense-fields.service.ts [hunk 1]
```diff
@@ -44,4 +45,88 @@ export class ExpenseFieldsService {
       )
     );
   }
+
+  getUserRoles() {
+    return from(this.authService.getRoles());
+  }
+
+  findCommonRoles(roles) {
```
[the remark this answers is not in the document]
Comment (Dimple16) [reply src/app/core/services/expense-fields.service.ts:61]: existing method moved from `transaction_field_configurations` service

File: src/app/core/services/expense-fields.service.ts [hunk 2]
```diff
@@ -52,4 +53,87 @@ export class ExpenseFieldsService {
       )
     );
   }
+
```
Comment (Dimple16) [inline src/app/core/services/expense-fields.service.ts:56]: All below methods are moved from `transaction_field_configurations` service. No changes done.

File: src/app/core/services/expense-fields.service.ts [hunk 3]
```diff
@@ -44,4 +45,88 @@ export class ExpenseFieldsService {
...
+        }
+
+        if (filteredField) {
+          filteredField.field = field;
+        }
+        return filteredField;
+      })
+        .filter(filteredField => !!filteredField)
+      ),
+      switchMap(fields => {
```
[the remark this answers is not in the document]
Comment (Dimple16) [reply src/app/core/services/expense-fields.service.ts:105]: existing method moved from transaction_field_configurations service - not changing existing ones

**Review by madhavmansuriya40 — APPROVED** [review]

File: src/app/core/models/v1/default-txn-field-values.model.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export interface DefaultTxnFieldValues {
+  bus_travel_class?: string;
```
Comment (abhishek1234321) [inline src/app/core/models/v1/default-txn-field-values.model.ts:2]: any reason on why all of these should be optional?

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 5]
```diff
@@ -1462,11 +1464,11 @@ export class AddEditExpensePage implements OnInit {
       map((tfcMap: any) => {
         if (tfcMap) {
```
Comment (abhishek1234321) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:1465]: let us change the variable from tfcMap to something else?

**Review by abhishek1234321 — CHANGES_REQUESTED** [review]

File: src/app/core/models/v1/default-txn-field-values.model.ts (hunk 4 above)
Comment (Dimple16) [reply src/app/core/models/v1/default-txn-field-values.model.ts:2]: while computing the default txn field values, we return only the fields which have a default value, these keys will not be present always

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts (hunk 5 above)
Comment (Dimple16) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts]: updated to `expenseFieldsMap`

**Review by abhishek1234321 — CHANGES_REQUESTED** [review]

File: src/app/core/services/expense-fields.service.ts [hunk 6]
```diff
@@ -22,7 +23,8 @@ export class ExpenseFieldsService {
         return this.apiService.get('/expense_fields', {
           params: {
             org_id: eou.ou.org_id,
-            is_enabled: true
+            is_enabled: true,
+            is_custom: false
```
[the remark this answers is not in the document]
Comment (Dimple16) [reply src/app/core/services/expense-fields.service.ts:27]: for expense fields we call the mapping method, custom fields do not have the mapping

**Review by abhishek1234321 — APPROVED** [review]

**Review by Julias0 — APPROVED** [review]

**Pull request closed without merging**